build.gradle

```diff
@@ -136,7 +143,8 @@ subprojects {
 
           [compileJava, compileTestJava].each() {
             // Protobuf-generated code produces some warnings.
-            it.options.compilerArgs += ["-Xlint:-cast"]
+            it.options.compilerArgs += ["-Xlint:-cast", "-Xep:MissingOverride:OFF",
+                "-Xep:ReferenceEquality:OFF", "-Xep:FunctionalInterfaceClash:OFF"]
```

Comment:
I'm not certain protobuf should fix the warnings. Depending on your perspective it could be blamed on protobuf for not being perfect in errorprone's eyes, protobuf-gradle-plugin for mixing generated and non-generated code in the JavaCompile task, gradle for making it hard for protobuf-gradle-plugin to use a separate compilation task, or errorprone for not supporting filtering. However, if I view the problem from each project's perspective, it doesn't seem like their problem and it would be unlikely they would change.